server/app/schemas/plan.py

```
@@ -0,0 +1,50 @@
+    stripe_price_id: Optional[str] = Field(None, description="Stripe price ID")
+    
+    # Per-rooftop pricing (new model)
+    credits_per_rooftop: int = Field(..., description="Number of credits included per rooftop per month")
+    price_per_rooftop: Decimal = Field(..., description="Monthly price per rooftop in USD")
+    overage_price_per_credit: Decimal = Field(..., description="Price per credit/call for overage usage")
+    
+    class Config:
+        """Pydantic config."""
+        json_encoders = {
+            Decimal: lambda v: str(v) if v else None,
+        }
```

Ruff (0.15.0)

[warning] Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)